# HG changeset patch
# User ohair
# Date 1305210277 25200
# Node ID 13fa9a0c628f58326cc376b7623a03b9d796136e
# Parent  caed82420c5d01f5aaee5b15e8e37befc8db3d49
7043921: generate java-rmi.cgi on 64 bit platform
Reviewed-by: omajid, katleman

diff -r caed82420c5d -r 13fa9a0c628f make/sun/rmi/rmi/Makefile
--- openjdk/jdk/make/sun/rmi/rmi/Makefile	Wed May 11 14:12:50 2011 -0700
+++ openjdk/jdk/make/sun/rmi/rmi/Makefile	Thu May 12 07:24:37 2011 -0700
@@ -85,16 +85,21 @@
 	sun.rmi.registry.RegistryImpl \
 	sun.rmi.transport.DGCImpl
 
-ifeq ($(PLATFORM), windows)
-build: stubs
-else # PLATFORM
-ifneq ($(ARCH_DATA_MODEL), 32)
-build: stubs
-else # ARCH_DATA_MODEL
-build: stubs bin
+#
+# The java-rmi.cgi script in bin/ only gets delivered in certain situations
+#
+BUILD_TARGETS = stubs
+ifeq ($(PLATFORM), linux)
+  BUILD_TARGETS += bin
 endif
+ifeq ($(PLATFORM), solaris)
+  ifeq ($(ARCH_DATA_MODEL), 32)
+    BUILD_TARGETS += bin
+  endif
 endif
 
+build: $(BUILD_TARGETS)
+
 clean clobber:: bin.clean
 
 
